engine/table/src/main/java/io/deephaven/engine/table/impl/sources/sparse/CharOneOrN.java

```diff
@@ -42,6 +50,41 @@ public void ensureIndex(int idx, SoftRecycler<Block1[]> recycler) {
             }
         }
 
+        /**
+         * Called by the {@link io.deephaven.engine.table.impl.sources.CharacterSparseArraySource} at the end of a cycle
+         * when blocks have been completely eliminated from the output rowset.  The input is not a row key in our address
+         * space, but rather an index for the block to clear (i.e. row key >> {@link SparseConstants#BLOCK2_SHIFT}).
+         *
+         * @param block2Idx the block index to clear
+         */
+        public void clearBlock(final long block2Idx) {
+            final int block0 = (int) (block2Idx >> (BLOCK0_SHIFT - BLOCK2_SHIFT)) & BLOCK0_MASK;
+            final int block1 = (int) (block2Idx >> (BLOCK1_SHIFT - BLOCK2_SHIFT)) & BLOCK1_MASK;
+            final int block2 = (int) (block2Idx) & BLOCK2_MASK;
+
+            final Block1 blocks0 = get(block0);
+            if (blocks0 == null) {
+                // we should not be asking to clear a block that does not actually exist
+                throw new IllegalStateException();
+            }
+
+            final Block2 blocks1 = blocks0.get(block1);
+            if (blocks1 == null) {
+                // we should not be asking to clear a block that does not actually exist
+                throw new IllegalStateException();
+            }
```

Comment:
I'm lost by the suffixes used here. For readability, shouldn't this be:
```
            final Block1 blocks1 = get(block0);
            if (blocks1 == null) {
                // we should not be asking to clear a block that does not actually exist
                throw new IllegalStateException();
            }
```
The logic looks correct, but the naming confuses.

Similarly for `Block2 blocks1`. Or maybe these should be `blocksFrom0`/`blocksFrom1` or something clearer? Not a big problem, but it looks like a logic error is happening when it isn't.